### ISSUE TITLE
problem with getComponents, getContents, getComponent

### ISSUE THREAD

**Issue by gregorysemah** [joseph/Monocle#50]
Hello,

I'm experimenting to extract epub files and put it in an automatic way in monocle with bookData object
Some parsing of epub files, according to your doc give me for example this:

<script language="javascript">
var bookData = {
  getComponents: function () {
    return [

```
                'title.xml',

                'about.xml',

                'main0.xml',

                'similar.xml',

                'feedbooks.xml',

];
```

  },
  getContents: function () {
    return [
        {title:"Title", src:"title.xml"},{title:"About", src:"about.xml"},{title:"The Vision Of Tom Chuff", src:"main0.xml"},{title:"Recommendations", src:"similar.xml"},
    ]
  },
  getComponent: function (componentId) {

```
                'title.xml':'<div class=\"body\"><div class=\"logo\"><img alt=\"Feedbooks\" src=\"/media/uploads/media_up/52/logo-feedbooks-tiny.png\" title=\"Feedbooks\"/></div><h1>The Vision Of Tom Chuff</h1><h2>Joseph Sheridan Le Fanu</h2></div><div class=\"infos\"><br/><b>Published:</b> 1870<br/><b>Categorie(s):</b> Fiction, Ghost, Horror, Short Stories<br/><b>Source:</b> http://gutenberg.org</div>',

                'about.xml':'<div class=\"body\"><div class=\"aboutauthor\"><strong>About Le Fanu:</strong><br/><p>Sheridan Le Fanu was born at No. 45 Lower Dominick Steet, Dublin, into a literary family of Huguenot origins. Both his grandmother Alicia Sheridan Le Fanu and his great-uncle Richard Brinsley Sheridan were playwrights. His niece Rhoda Broughton would become a very successful novelist. Within a year of his birth his family moved to the Royal Hibernian Military School in Phoenix Park, where his father, an Anglican clergyman, was the chaplain of the establishment. Phoenix Park and the adjacent village and parish church of Chapelizod were to feature in Le Fanu\'s later stories. Le Fanu studied law at Trinity College in Dublin, where he was elected Auditor of the College Historical Society. He was called to the bar in 1839, but he never practised and soon abandoned law for journalism. In 1838 he began contributing stories to the Dublin University Magazine, including his first ghost story, entitled &quot;A Strange Event in the Life of Schalken the Painter&quot; (1839). He became owner of several newspapers from 1840, including the Dublin Evening Mail and the Warder. In 1844 Le Fanu married Susanna Bennett, the daughter of a leading Dublin barrister. In 1847 he supported John Mitchell and Thomas Meagher in their campaign against the indifference of the Government to the Irish Famine. His support cost him the nomination as Tory MP for County Carlow in 1852. His personal life also became difficult at this time, as his wife Susanna suffered from increasing neurotic symptoms. She died in 1858 in unclear circumstances, and anguished excerpts from Le Fanu\'s diaries suggest that he felt guilt as well as loss. However, it was only after her death that, becoming something of a recluse, he devoted himself full time to writing. In 1861 he became the editor and proprietor of the Dublin University Magazine and he began exploiting double exposure: serializing in the Dublin University Magazine and then revising for the English market. The House by the Churchyard and Wylder\'s Hand were both published in this way. After the lukewarm reviews of the former novel, set in the Phoenix Park area of Dublin, Le Fanu signed a contract with Richard Bentley, his London publisher, which specified that future novels be stories &quot;of an English subject and of modern times&quot;, a step Bentley thought necessary in order for Le Fanu to satisfy the English audience. Le Fanu succeeded in this aim in 1864, with the publication of Uncle Silas, which he set in Derbyshire. In his very last short stories, however, Le Fanu returned to Irish folklore as an inspiration and encouraged his friend Patrick Kennedy to contribute folklore to the D.U.M. Le Fanu died in his native Dublin on February 7, 1873. Today there is a road in Ballyfermot, near his childhood home in south-west Dublin, named after him. Source: Wikipedia</p></div><div class=\"also\"><strong>Also available on Feedbooks Le Fanu:</strong><ul><li><i><a href=\"http://generation.feedbooks.com/book/2834.epub\">A Stable for Nightmares</a></i> (1896)</li><li><i><a href=\"http://generation.feedbooks.com/book/1506.epub\">Carmilla</a></i> (1871)</li><li><i><a href=\"http://generation.feedbooks.com/book/1505.epub\">Uncle Silas</a></i> (1864)</li><li><i><a href=\"http://generation.feedbooks.com/book/1575.epub\">The Child That Went With The Fairies</a></i> (1870)</li><li><i><a href=\"http://generation.feedbooks.com/book/1569.epub\">An Account of Some Strange Disturbances in Aungier Street</a></i> (1853)</li><li><i><a href=\"http://generation.feedbooks.com/book/1570.epub\">An Authentic Narrative of a Haunted House</a></i> (1862)</li><li><i><a href=\"http://generation.feedbooks.com/book/1579.epub\">Ghost Stories of Chapelizod</a></i> (1851)</li><li><i><a href=\"http://generation.feedbooks.com/book/1589.epub\">The House by the Church-Yard</a></i> (1863)</li><li><i><a href=\"http://generation.feedbooks.com/book/1529.epub\">The Evil Guest</a></i> (1851)</li><li><i><a href=\"http://generation.feedbooks.com/book/1583.epub\">The Mysterious Lodger</a></i> (1850)</li></ul></div><div class=\"also\"><strong>Note:</strong> This book is brought to you by Feedbooks<br/><a href=\"http://www.feedbooks.com\">http://www.feedbooks.com</a><br/> Strictly for personal use, do not use this file for commercial purposes.</div></div>',

                'main0.xml':'<div class=\"body\"><div class=\"text\"><p>At the edge of melancholy Catstean Moor, in the north of England, with half-a-dozen ancient poplar-trees with rugged and hoary stems around, one smashed across the middle by a flash of lightning thirty summers before, and all by their great height dwarfing the abode near which they stand, there squats a rude stone house, with a thick chimney, a kitchen and bedroom on the ground-floor, and a loft, accessible by a ladder, under the shingle roof, divided into two rooms.</p><p>Its owner was a man of ill repute. Tom Chuff was his name. A shock-headed, broad-shouldered, powerful man, though somewhat short, with lowering brows and a sullen eye. He was a poacher, and hardly made an ostensible pretence of earning his bread by any honest industry. He was a drunkard. He beat his wife, and led his children a life of terror and lamentation, when he was at home. It was a blessing to his frightened little family when he absented himself, as he sometimes did, for a week or more together.</p><p>On the night I speak of he knocked at the door with his cudgel at about eight o\'clock. It was winter, and the night was very dark. Had the summons been that of a bogie from the moor, the inmates of this small house could hardly have heard it with greater terror.</p><p>His wife unbarred the door in fear and haste. Her hunchbacked sister stood by the hearth, staring toward the threshold. The children cowered behind.</p><p>Tom Chuff entered with his cudgel in his hand, without speaking, and threw himself into a chair opposite the fire. He had been away two or three days. He looked haggard, and his eyes were bloodshot. They knew he had been drinking.</p><p>Tom raked and knocked the peat fire with his stick, and thrust his feet close to it. He signed towards the little dresser, and nodded to his wife, and she knew he wanted a cup, which in silence she gave him. He pulled a bottle of gin from his coat-pocket, and nearly filling the teacup, drank off the dram at a few gulps.</p><p>He usually refreshed himself with two or three drams of this kind before beating the inmates of his house. His three little children, cowering in a corner, eyed him from under a table, as Jack did the ogre in the nursery tale. His wife, Nell, standing behind a chair, which she was ready to snatch up to meet the blow of the cudgel, which might be levelled at her at any moment, never took her eyes off him; and hunchbacked Mary showed the whites of a large pair of eyes, similarly employed, as she stood against the oaken press, her dark face hardly distinguishable in the distance from the brown panel behind it.</p><p>Tom Chuff was at his third dram, and had not yet spoken a word since his entrance, and the suspense was growing dreadful, when, on a sudden, he leaned back in his rude seat, the cudgel slipped from his hand, a change and a death-like pallor came over his face.</p><p>For a while they all stared on; such was their fear of him, they dared not speak or move, lest it should prove to have been but a doze, and Tom should wake up and proceed forthwith to gratify his temper and exercise his cudgel.</p><p>In a very little time, however, things began to look so odd, that they ventured, his wife and Mary, to exchange glances full of doubt and wonder. He hung so much over the side of the chair, that if it had not been one of cyclopean clumsiness and weight, he would have borne it to the floor. A leaden tint was darkening the pallor of his face. They were becoming alarmed, and finally braving everything his wife timidly said, &quot;Tom!&quot; and then more sharply repeated it, and finally cried the appellative loudly, and again and again, with the terrified accompaniment, &quot;He\'s dying—he\'s dying!&quot; her voice rising to a scream, as she found that neither it nor her plucks and shakings of him by the shoulder had the slightest effect in recalling him from his torpor.</p><p>And now from sheer terror of a new kind the children added their shrilly piping to the talk and cries of their seniors; and if anything could have called Tom up from his lethargy, it might have been the piercing chorus that made the rude chamber of the poacher\'s habitation ring again. But Tom continued unmoved, deaf, and stirless.</p><p>His wife sent Mary down to the village, hardly a quarter of a mile away, to implore of the doctor, for whose family she did duty as laundress, to come down and look at her husband, who seemed to be dying.</p><p>The doctor, who was a good-natured fellow, arrived. With his hat still on, he looked at Tom, examined him, and when he found that the emetic he had brought with him, on conjecture from Mary\'s description, did not act, and that his lancet brought no blood, and that he felt a pulseless wrist, he shook his head, and inwardly thought:</p><p>&quot;What the plague is the woman crying for? Could she have desired a greater blessing for her children and herself than the very thing that has happened?&quot;</p><p>Tom, in fact, seemed quite gone. At his lips no breath was perceptible. The doctor could discover no pulse. His hands and feet were cold, and the chill was stealing up into his body.</p><p>The doctor, after a stay of twenty minutes, had buttoned up his great-coat again and pulled down his hat, and told Mrs. Chuff that there was no use in his remaining any longer, when, all of a sudden, a little rill of blood began to trickle from the lancet-cut in Tom Chuffs temple.</p><p>&quot;That\'s very odd,&quot; said the doctor. &quot;Let us wait a little.&quot;</p><p>I must describe now the sensations which Tom Chuff had experienced.</p><p>With his elbows on his knees, and his chin upon his hands, he was staring into the embers, with his gin beside him, when suddenly a swimming came in his head, he lost sight of the fire, and a sound like one stroke of a loud church bell smote his brain.</p><p>Then he heard a confused humming, and the leaden weight of his head held him backward as he sank in his chair, and consciousness quite forsook him.</p><p>When he came to himself he felt chilled, and was leaning against a huge leafless tree. The night was moonless, and when he looked up he thought he had never seen stars so large and bright, or sky so black. The stars, too, seemed to blink down with longer intervals of darkness, and fiercer and more dazzling emergence, and something, he vaguely thought, of the character of silent menace and fury.</p><p>He had a confused recollection of having come there, or rather of having been carried along, as if on men\'s shoulders, with a sort of rushing motion. But it was utterly indistinct; the imperfect recollection simply of a sensation. He had seen or heard nothing on his way.</p><p>He looked round. There was not a sign of a living creature near. And he began with a sense of awe to recognise the place.</p><p>The tree against which he had been leaning was one of the noble old beeches that surround at irregular intervals the churchyard of Shackleton, which spreads its green and wavy lap on the edge of the Moor of Catstean, at the opposite side of which stands the rude cottage in which he had just lost consciousness. It was six miles or more across the moor to his habitation, and the black expanse lay before him, disappearing dismally in the darkness. So that, looking straight before him, sky and land blended together in an undistinguishable and awful blank.</p><p>There was a silence quite unnatural over the place. The distant murmur of the brook, which he knew so well, was dead; not a whisper in the leaves about him; the air, earth, everything about and above was indescribably still; and he experienced that quaking of the heart that seems to portend the approach of something awful. He would have set out upon his return across the moor, had he not an undefined presentiment that he was waylaid by something he dared not pass.</p><p>The old grey church and tower of Shackleton stood like a shadow in the rear. His eye had grown accustomed to the obscurity, and he could just trace its outline. There were no comforting associations in his mind connected with it; nothing but menace and misgiving. His early training in his lawless calling was connected with this very spot. Here his father used to meet two other poachers, and bring his son, then but a boy, with him.</p><p>Under the church porch, towards morning, they used to divide the game they had taken, and take account of the sales they had made on the previous day, and make partition of the money, and drink their gin. It was here he had taken his early lessons in drinking, cursing, and lawlessness. His father\'s grave was hardly eight steps from the spot where he stood. In his present state of awful dejection, no scene on earth could have so helped to heighten his fear.</p><p>There was one object close by which added to his gloom. About a yard away, in rear of the tree, behind himself, and extending to his left, was an open grave, the mould and rubbish piled on the other side. At the head of this grave stood the beech-tree; its columnar stem rose like a huge monumental pillar. He knew every line and crease on its smooth surface. The initial letters of his own name, cut in its bark long ago, had spread out and wrinkled like the grotesque capitals of a fanciful engraver, and now with a sinister significance overlooked the open grave, as if answering his mental question, &quot;Who for is t\' grave cut?&quot;</p><p>He felt still a little stunned, and there was a faint tremor in his joints that disinclined him to exert himself; and, further, he had a vague apprehension that take what direction he might, there was danger around him worse than that of staying where he was.</p><p>On a sudden the stars began to blink more fiercely, a faint wild light overspread for a minute the bleak landscape, and he saw approaching from the moor a figure at a kind of swinging trot, with now and then a zig-zag hop or two, such as men accustomed to cross such places make, to avoid the patches of slob or quag that meet them here and there. This figure resembled his father\'s, and like him, whistled through his finger by way of signal as he approached; but the whistle sounded not now shrilly and sharp, as in old times, but immensely far away, and seemed to sing strangely through Tom\'s head. From habit or from fear, in answer to the signal, Tom whistled as he used to do five-and-twenty years ago and more, although he was already chilled with an unearthly fear.</p><p>Like his father, too, the figure held up the bag that was in his left hand as he drew near, when it was his custom to call out to him what was in it. It did not reassure the watcher, you may be certain, when a shout unnaturally faint reached him, as the phantom dangled the bag in the air, and he heard with a faint distinctness the words, &quot;Tom Chuff\'s soul!&quot;</p><p>Scarcely fifty yards away from the low churchyard fence at which Tom was standing, there was a wider chasm in the peat, which there threw up a growth of reeds and bulrushes, among which, as the old poacher used to do on a sudden alarm, the approaching figure suddenly cast itself down.</p><p>From the same patch of tall reeds and rushes emerged instantaneously what he at first mistook for the same figure creeping on all-fours, but what he soon perceived to be an enormous black dog with a rough coat like a bear\'s, which at first sniffed about, and then started towards him in what seemed to be a sportive amble, bouncing this way and that, but as it drew near it displayed a pair of fearful eyes that glowed like live coals, and emitted from the monstrous expanse of its jaws a terrifying growl.</p><p>This beast seemed on the point of seizing him, and Tom recoiled in panic and fell into the open grave behind him. The edge which he caught as he tumbled gave way, and down he went, expecting almost at the same instant to reach the bottom. But never was such a fall! Bottomless seemed the abyss! Down, down, down, with immeasurable and still increasing speed, through utter darkness, with hair streaming straight upward, breathless, he shot with a rush of air against him, the force of which whirled up his very arms, second after second, minute after minute, through the chasm downward he flew, the icy perspiration of horror covering his body, and suddenly, as he expected to be dashed into annihilation, his descent was in an instant arrested with a tremendous shock, which, however, did not deprive him of consciousness even for a moment.</p><p>He looked about him. The place resembled a smoke-stained cavern or catacomb, the roof of which, except for a ribbed arch here and there faintly visible, was lost in darkness. From several rude passages, like the galleries of a gigantic mine, which opened from this centre chamber, was very dimly emitted a dull glow as of charcoal, which was the only light by which he could imperfectly discern the objects immediately about him.</p><p>What seemed like a projecting piece of the rock, at the corner of one of these murky entrances, moved on a sudden, and proved to be a human figure, that beckoned to him. He approached, and saw his father. He could barely recognise him, he was so monstrously altered.</p><p>&quot;I\'ve been looking for you, Tom. Welcome home, lad; come along to your place.&quot;</p><p>Tom\'s heart sank as he heard these words, which were spoken in a hollow and, he thought, derisive voice that made him tremble. But he could not help accompanying the wicked spirit, who led him into a place, in passing which he heard, as it were from within the rock, deadful cries and appeals for mercy.</p><p>&quot;What is this?&quot; said he.</p><p>&quot;Never mind.&quot;</p><p>&quot;Who are they?&quot;</p><p>&quot;New-comers, like yourself, lad,&quot; answered his father apathetically. &quot;They give over that work in time, finding it is no use.&quot;</p><p>&quot;What shall I do?&quot; said Tom, in an agony.</p><p>&quot;It\'s all one.&quot;</p><p>&quot;But what shall I do?&quot; reiterated Tom, quivering in every joint and nerve.</p><p>&quot;Grin and bear it, I suppose.&quot;</p><p>&quot;For God\'s sake, if ever you cared for me, as I am your own child, let me out of this!&quot;</p><p>&quot;There\'s no way out.&quot;</p><p>&quot;If there\'s a way in there\'s a way out, and for Heaven\'s sake let me out of this.&quot;</p><p>But the dreadful figure made no further answer, and glided backwards by his shoulder to the rear; and others appeared in view, each with a faint red halo round it, staring on him with frightful eyes, images, all in hideous variety, of eternal fury or derision. He was growing mad, it seemed, under the stare of so many eyes, increasing in number and drawing closer every moment, and at the same time myriads and myriads of voices were calling him by his name, some far away, some near, some from one point, some from another, some from behind, close to his ears. These cries were increased in rapidity and multitude, and mingled with laughter, with flitting blasphemies, with broken insults and mockeries, succeeded and obliterated by others, before he could half catch their meaning.</p><p>All this time, in proportion to the rapidity and urgency of these dreadful sights and sounds, the epilepsy of terror was creeping up to his brain, and with a long and dreadful scream he lost consciousness.</p><p>When he recovered his senses, he found himself in a small stone chamber, vaulted above, and with a ponderous door. A single point of light in the wall, with a strange brilliancy illuminated this cell.</p><p>Seated opposite to him was a venerable man with a snowy beard of immense length; an image of awful purity and severity. He was dressed in a coarse robe, with three large keys suspensed from his girdle. He might have filled one\'s idea of an ancient porter of a city gate; such spiritual cities, I should say, as John Bunyan loved to describe.</p><p>This old man\'s eyes were brilliant and awful, and fixed on him as they were, Tom Chuff felt himself helplessly in his power. At length he spoke:</p><p>&quot;The command is given to let you forth for one trial more. But if you are found again drinking with the drunken, and beating your fellow-servants, you shall return through the door by which you came, and go out no more.&quot;</p><p>With these words the old man took him by the wrist and led him through the first door, and then unlocking one that stood in the cavern outside, he struck Tom Chuff sharply on the shoulder, and the door shut behind him with a sound that boomed peal after peal of thunder near and far away, and all round and above, till it rolled off gradually into silence. It was totally dark, but there was a fanning of fresh cool air that overpowered him. He felt that he was in the upper world again.</p><p>In a few minutes he began to hear voices which he knew, and first a faint point of light appeared before his eyes, and gradually he saw the flame of the candle, and, after that, the familiar faces of his wife and children, and he heard them faintly when they spoke to him, although he was as yet unable to answer.</p><p>He also saw the doctor, like an isolated figure in the dark, and heard him say:</p><p>&quot;There, now, you have him back. He\'ll do, I think.&quot;</p><p>His first words, when he could speak and saw clearly all about him, and felt the blood on his neck and shirt, were:</p><p>&quot;Wife, forgie me. I\'m a changed man. Send for\'t sir.&quot;</p><p>Which last phrase means, &quot;Send for the clergyman.&quot;</p><p>When the vicar came and entered the little bedroom where the scared poacher, whose soul had died within him, was lying, still sick and weak, in his bed, and with a spirit that was prostrate with terror, Tom Chuff feebly beckoned the rest from the room, and, the door being closed, the good parson heard the strange confession, and with equal amazement the man\'s earnest and agitated vows of amendment, and his helpless appeals to him for support and counsel.</p><p>These, of course, were kindly met; and the visits of the rector, for some time, were frequent.</p><p>One day, when he took Tom Chuff\'s hand on bidding him good-bye, the sick man held it still, and said:</p><p>&quot;Ye\'r vicar o\' Shackleton, sir, and if I sud dee, ye\'ll promise me a\'e thing, as I a promised ye a many. I a said I\'ll never gie wife, nor barn, nor folk o\' no sort, skelp nor sizzup more, and ye\'ll know o\' me no more among the sipers. Nor never will Tom draw trigger, nor set a snare again, but in an honest way, and after that ye\'ll no make it a bootless bene for me, but bein\', as I say, vicar o\' Shackleton, and able to do as ye list, ye\'ll no let them bury me within twenty good yerd-wands measure o\' the a\'d beech trees that\'s round the churchyard of Shackleton.&quot;</p><p>&quot;I see; you would have your grave, when your time really comes, a good way from the place where lay the grave you dreamed of.&quot;</p><p>&quot;That\'s jest it. I\'d lie at the bottom o\' a marl-pit liefer! And I\'d be laid in anither churchyard just to be shut o\' my fear o\' that, but that a\' my kinsfolk is buried beyond in Shackleton, and ye\'ll gie me yer promise, and no break yer word.&quot;</p><p>&quot;I do promise, certainly. I\'m not likely to outlive you; but, if I should, and still be vicar of Shackleton, you shall be buried somewhere as near the middle of the churchyard as we can find space.&quot;</p><p>&quot;That\'ll do.&quot;</p><p>And so content they parted.</p><p>The effect of the vision upon Tom Chuff was powerful, and promised to be lasting. With a sore effort he exchanged his life of desultory adventure and comparative idleness for one of regular industry. He gave up drinking; he was as kind as an originally surly nature would allow to his wife and family; he went to church; in fine weather they crossed the moor to Shackleton Church; the vicar said he came there to look at the scenery of his vision, and to fortify his good resolutions by the reminder.</p><p>Impressions upon the imagination, however, are but transitory, and a bad man acting under fear is not a free agent; his real character does not appear. But as the images of the imagination fade, and the action of fear abates, the essential qualities of the man reassert themselves.</p><p>So, after a time, Tom Chuff began to grow weary of his new life; he grew lazy, and people began to say that he was catching hares, and pursuing his old contraband way of life, under the rose.</p><p>He came home one hard night, with signs of the bottle in his thick speech and violent temper. Next day he was sorry, or frightened, at all events repentant, and for a week or more something of the old horror returned, and he was once more on his good behaviour. But in a little time came a relapse, and another repentance, and then a relapse again, and gradually the return of old habits and the flooding in of all his old way of life, with more violence and gloom, in proportion as the man was alarmed and exasperated by the remembrance of his despised, but terrible, warning.</p><p>With the old life returned the misery of the cottage. The smiles, which had begun to appear with the unwonted sunshine, were seen no more. Instead, returned to his poor wife\'s face the old pale and heartbroken look. The cottage lost its neat and cheerful air, and the melancholy of neglect was visible. Sometimes at night were overheard, by a chance passer-by, cries and sobs from that ill-omened dwelling. Tom Chuff was now often drunk, and not very often at home, except when he came in to sweep away his poor wife\'s earnings.</p><p>Tom had long lost sight of the honest old parson. There was shame mixed with his degradation. He had grace enough left when he saw the thin figure of &quot;t\' sir&quot; walking along the road to turn out of his way and avoid meeting him. The clergyman shook his head, and sometimes groaned, when his name was mentioned. His horror and regret were more for the poor wife than for the relapsed sinner, for her case was pitiable indeed.</p><p>Her brother, Jack Everton, coming over from Hexley, having heard stories of all this, determined to beat Tom, for his ill-treatment of his sister, within an inch of his life. Luckily, perhaps, for all concerned, Tom happened to be away upon one of his long excursions, and poor Nell besought her brother, in extremity of terror, not to interpose between them. So he took his leave and went home muttering and sulky.</p><p>Now it happened a few months later that Nelly Chuff fell sick. She had been ailing, as heartbroken people do, for a good while. But now the end had come.</p><p>There was a coroner\'s inquest when she died, for the doctor had doubts as to whether a blow had not, at least, hastened her death. Nothing certain, however, came of the inquiry. Tom Chuff had left his home more than two days before his wife\'s death. He was absent upon his lawless business still when the coroner had held his quest.</p><p>Jack Everton came over from Hexley to attend the dismal obsequies of his sister. He was more incensed than ever with the wicked husband, who, one way or other, had hastened Nelly\'s death. The inquest had closed early in the day. The husband had not appeared.</p><p>An occasional companion—perhaps I ought to say accomplice—of Chuff\'s happened to turn up. He had left him on the borders of Westmoreland, and said he would probably be home next day. But Everton affected not to believe it. Perhaps it was to Tom Chuff, he suggested, a secret satisfaction to crown the history of his bad married life with the scandal of his absence from the funeral of his neglected and abused wife.</p><p>Everton had taken on himself the direction of the melancholy preparations. He had ordered a grave to be opened for his sister beside her mother\'s, in Shackleton churchyard, at the other side of the moor. For the purpose, as I have said, of marking the callous neglect of her husband, he determined that the funeral should take place that night. His brother Dick had accompanied him, and they and his sister, with Mary and the children, and a couple of the neighbours, formed the humble cortège.</p><p>Jack Everton said he would wait behind, on the chance of Tom Chuff coming in time, that he might tell him what had happened, and make him cross the moor with him to meet the funeral. His real object, I think, was to inflict upon the villain the drubbing he had so long wished to give him. Anyhow, he was resolved, by crossing the moor, to reach the churchyard in time to anticipate the arrival of the funeral, and to have a few words with the vicar, clerk, and sexton, all old friends of his, for the parish of Shackleton was the place of his birth and early recollections.</p><p>But Tom Chuff did not appear at his house that night. In surly mood, and without a shilling in his pocket, he was making his way homeward. His bottle of gin, his last investment, half emptied, with its neck protruding, as usual on such returns, was in his coat-pocket.</p><p>His way home lay across the moor of Catstean, and the point at which he best knew the passage was from the churchyard of Shackleton. He vaulted the low wall that forms its boundary, and strode across the graves, and over many a flat, half-buried tombstone, toward the side of the churchyard next Catstean Moor.</p><p>The old church of Shackleton and its tower rose, close at his right, like a black shadow against the sky. It was a moonless night, but clear. By this time he had reached the low boundary wall, at the other side, that overlooks the wide expanse of Catstean Moor. He stood by one of the huge old beech-trees, and leaned his back to its smooth trunk. Had he ever seen the sky look so black, and the stars shine out and blink so vividly? There was a deathlike silence over the scene, like the hush that precedes thunder in sultry weather. The expanse before him was lost in utter blackness. A strange quaking unnerved his heart. It was the sky and scenery of his vision! The same horror and misgiving. The same invincible fear of venturing from the spot where he stood. He would have prayed if he dared. His sinking heart demanded a restorative of some sort, and he grasped the bottle in his coat-pocket. Turning to his left, as he did so, he saw the piled-up mould of an open grave that gaped with its head close to the base of the great tree against which he was leaning.</p><p>He stood aghast. His dream was returning and slowly enveloping him. Everything he saw was weaving itself into the texture of his vision. The chill of horror stole over him.</p><p>A faint whistle came shrill and clear over the moor, and he saw a figure approaching at a swinging trot, with a zig-zag course, hopping now here and now there, as men do over a surface where one has need to choose their steps. Through the jungle of reeds and bulrushes in the foreground this figure advanced; and with the same unaccountable impulse that had coerced him in his dream, he answered the whistle of the advancing figure.</p><p>On that signal it directed its course straight toward him. It mounted the low wall, and, standing there, looked into the graveyard.</p><p>&quot;Who med answer?&quot; challenged the new-comer from his post of observation.</p><p>&quot;Me,&quot; answered Tom.</p><p>&quot;Who are you?&quot; repeated the man upon the wall.</p><p>&quot;Tom Chuff; and who\'s this grave cut for?&quot; He answered in a savage tone, to cover the secret shudder of his panic.</p><p>&quot;I\'ll tell you that, ye villain!&quot; answered the stranger, descending from the wall, &quot;I a\' looked for you far and near, and waited long, and now you\'re found at last.&quot;</p><p>Not knowing what to make of the figure that advanced upon him, Tom Chuff recoiled, stumbled, and fell backward into the open grave. He caught at the sides as he fell, but without retarding his fall.</p><p>An hour later, when lights came with the coffin, the corpse of Tom Chuff was found at the bottom of the grave. He had fallen direct upon his head, and his neck was broken. His death must have been simultaneous with his fall. Thus far his dream was accomplished.</p><p>It was his brother-in-law who had crossed the moor and approached the churchyard of Shackleton, exactly in the line which the image of his father had seemed to take in his strange vision. Fortunately for Jack Everton, the sexton and clerk of Shackleton church were, unseen by him, crossing the churchyard toward the grave of Nelly Chuff, just as Tom the poacher stumbled and fell. Suspicion of direct violence would otherwise have inevitably attached to the exasperated brother. As it was, the catastrophe was followed by no legal consequences.</p><p>The good vicar kept his word, and the grave of Tom Chuff is still pointed out by the old inhabitants of Shackleton pretty nearly in the centre of the churchyard. This conscientious compliance with the entreaty of the panic-stricken man as to the place of his sepulture gave a horrible and mocking emphasis to the strange combination by which fate had defeated his precaution, and fixed the place of his death.</p><p>The story was for many a year, and we believe still is, told round many a cottage hearth, and though it appeals to what many would term superstition, it yet sounded, in the ears of a rude and simple audience, a thrilling, and let us hope, not altogether fruitless homily.</p></div></div>',

                'similar.xml':'<div class=\"body\"><h3 style=\"text-align: center;\">Loved this book ?<br/> Similar users also downloaded</h3><ul class=\"similar\"><li class=\"author\">Joseph Sheridan Le Fanu</li><li><hr/><a href=\"http://generation.feedbooks.com/book/1574.epub\"><em>Wicked Captain Walshawe, Of Wauling</em></a></li></ul><ul class=\"similar\"><li class=\"author\">Joseph Sheridan Le Fanu</li><li><hr/><a href=\"http://generation.feedbooks.com/book/1571.epub\"><em>Ultor De Lacy: A Legend of Cappercullen</em></a></li></ul><ul class=\"similar\"><li class=\"author\">Joseph Sheridan Le Fanu</li><li><hr/><a href=\"http://generation.feedbooks.com/book/1580.epub\"><em>The Drunkard\'s Dream</em></a></li></ul><ul class=\"similar\"><li class=\"author\">Joseph Sheridan Le Fanu</li><li><hr/><a href=\"http://generation.feedbooks.com/book/1573.epub\"><em>Laura Silver Bell</em></a></li></ul><ul class=\"similar\"><li class=\"author\">Joseph Sheridan Le Fanu</li><li><hr/><a href=\"http://generation.feedbooks.com/book/1568.epub\"><em>Schalken the Painter</em></a></li></ul><ul class=\"similar\"><li class=\"author\">Joseph Sheridan Le Fanu</li><li><hr/><a href=\"http://generation.feedbooks.com/book/1576.epub\"><em>Stories Of Lough Guir</em></a></li></ul><ul class=\"similar\"><li class=\"author\">Joseph Sheridan Le Fanu</li><li><hr/><a href=\"http://generation.feedbooks.com/book/1572.epub\"><em>The Haunted Baronet</em></a></li></ul><ul class=\"similar\"><li class=\"author\">Joseph Sheridan Le Fanu</li><li><hr/><a href=\"http://generation.feedbooks.com/book/1643.epub\"><em>The Room in the Dragon Volant</em></a></li></ul><ul class=\"similar\"><li class=\"author\">Joseph Sheridan Le Fanu</li><li><hr/><a href=\"http://generation.feedbooks.com/book/1553.epub\"><em>Madam Crowl\'s Ghost and the Dead Sexton</em></a></li></ul><ul class=\"similar\"><li class=\"author\">Joseph Sheridan Le Fanu</li><li><hr/><a href=\"http://generation.feedbooks.com/book/1582.epub\"><em>The Ghost and the Bone-setter</em></a></li></ul></div>',

                'feedbooks.xml':'<div class=\"body\"><div class=\"logo2\"><img alt=\"FeedBooks\" src=\"/media/uploads/media_up/52/logo-feedbooks.png\" title=\"FeedBooks\"/></div><div class=\"texttitle\"><b>www.feedbooks.com</b></div><div style=\"text-align: center;\">Food for the mind</div></div>',

}[componentId];
```

  },
  getMetaData: function(key) {
    return {
      title: "A book",
      creator: "Inventive Labs"
    }[key];
  }
}
//Monocle.Events.listen(window, 'load', function () {Monocle.Reader('rdr','', { panels: Monocle.Panels.TwoPane })});

Monocle.Events.listen(window, 'load', function () {
    Monocle.Reader('rdr',bookData,{panels: Monocle.Panels.TwoPane});
});

CHROMIUM give me the error "Uncaught SyntaxError: Unexpected token :"
I think this is the ":" of the getComponent which separate foo.xml to the content of the xml itself

So i tried the other way, according to your doc:

If i don't want to use bookData , I put the right part of each iteration of getComponent directly in 'rdr' div as follow

<div id="rdr" style="width: 500px; height: 500px;">
blablabla...
</div>

But the problem is that I can't browse more than 2 pages and no errors are shown....

After 3 days of investigation, I'm a bit stuck...

Do you have any ideas of what's going on?

Really thanks in advance!

Greg
